Remove the timeout in async_test for missing tests

qiuzhong opened the pull request:
Some async_test tests in webrtc and xhr with timeout are missed during
removing all the timeout in async_test. As part of goal #11120, this tests
should be updated too.

qiuzhong committed:
Remove the timeout in async_test for webrtc and xhr tests

Some async_test tests in webrtc and xhr with timeout are missed during
removing all the timeout in async_test. As part of goal #11120, this tests
should be updated too.

qiuzhong committed:
Remove timeout in async_test for geolocation-API tests

This async_test was missed during removing all the timeout from async_test.
Add `<meta name=timeout content=long>` in the html head could probably fix
this.
Related: #14118

foolip commented on xhr/abort-during-upload.htm:
```diff
@@ -11,7 +11,7 @@
     <div id="log"></div>
     <script>
-      var test = async_test(document.title, {timeout:1100})
+      var test = async_test(document.title);
```
```suggestion
      var test = async_test()
```

foolip replied:
`document.title` is the default test name if none is given, so this can be removed.

foolip commented on webrtc/simplecall.https.html:
```diff
@@ -17,7 +17,7 @@
   <script src="/resources/testharnessreport.js"></script>
   <script type="text/javascript">
-  var test = async_test('Can set up a basic WebRTC call.', {timeout: 5000});
+  var test = async_test('Can set up a basic WebRTC call.');
```
Taskcluster is failing on Firefox because this test crashed 1/10 times. That's hardly because of these changes though.

foolip reviewed (approved):
LGTM % nit with suggested fix

foolip committed:
Update xhr/abort-during-upload.htm

Remove `document.title` default argument from `async_test` as `document.title` is effectively already used
in testharness.js

Co-Authored-By: qiuzhong <zhongx.qiu@intel.com>
